src/MSALWrapper.Test/LockedTest.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace MSALWrapper.Test
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using FluentAssertions;
+
+    using Microsoft.Authentication.MSALWrapper;
+    using Microsoft.Extensions.Logging;
+
+    using Moq;
+
+    using NUnit.Framework;
+
+    internal class LockedTest
+    {
+        private static readonly TimeSpan TenSec = TimeSpan.FromSeconds(10);
+
+        private Mock<ILogger> mockLogger;
+
+        [SetUp]
+        public void SetUp()
+        {
+            this.mockLogger = new Mock<ILogger>();
+        }
+
+        [Test]
+        public void Get_Int()
+        {
+            var lockName = "get_an_int";
+            var subject = Locked.Execute(this.mockLogger.Object, lockName, TenSec, () => Task.FromResult(42));
+            subject.Should().Be(42);
+        }
+
+        [Test]
+        public void Get_String()
+        {
+            var lockName = "get_a_string";
+            var subject = Locked.Execute(this.mockLogger.Object, lockName, TenSec, () => Task.FromResult("hi there"));
+            subject.Should().Be("hi there");
+        }
+
+        [Test]
+        public void Execute_TimeOut()
+        {
+            var lockName = "short task times out while long task is running";
+            var shortTimeOut = TimeSpan.FromMilliseconds(1);
+
+            AutoResetEvent hasLock = new AutoResetEvent(false);
+            AutoResetEvent assertionMade = new AutoResetEvent(false);
+
+            Func<int> longFunc = () => Locked.Execute(this.mockLogger.Object, lockName, TenSec, () =>
+            {
+                // Signal that we have the lock, and wait for the assertion to be made.
+                hasLock.Set();
+                assertionMade.WaitOne();
+                return Task.FromResult(42);
+            });
+
+            Action subject = () => Locked.Execute(this.mockLogger.Object, lockName, shortTimeOut, () => Task.FromResult(0));
+
+            // Start longFunc, and wait for the lock to be acquired
+            Task<int> longTask = Task.Run(longFunc);
+            hasLock.WaitOne();
+            subject.Should().Throw<TimeoutException>();

```

Comment:
We can also assert the timeout message here?

Reply:
Good call out, we can make an assertion about the Exception Message here.

Reply:
Assertion added.